crates/order-validation/src/banned.rs

```diff
@@ -53,28 +53,60 @@ impl Users {
         }
     }
 
-    /// Returns `true` if the given address is banned or an error if the result
-    /// cannot be determined.
-    pub async fn is_banned(&self, address: H160) -> Result<bool, Error> {
-        if self.list.contains(&address) {
-            return Ok(true);
-        }
+    /// Returns a subset of addresses from the input iterator which are banned.
+    pub async fn banned(&self, addresses: impl Iterator<Item = H160>) -> HashSet<H160> {
```

Comment:
nanonit: In my experience it's usually more convenient to expect `impl IntoIterator`. That way you can pass iterators, slices and collections without having to call `.into_iter()`.
```suggestion
    pub async fn banned(&self, addresses: impl IntoIterator<Item = H160>) -> HashSet<H160> {
```